nixos/modules/services/web-apps/audiobookshelf.nix

```diff
@@ -0,0 +1,113 @@
+{ pkgs
+, lib
+, config
+, ...
+}:
+
+with lib; let
+  cfg = config.services.audiobookshelf;
+in
+
+{
+  options.services.audiobookshelf = {
+    enable = mkEnableOption (lib.mdDoc  "Audiobookshelf, self-hosted audiobook and podcast server");
+
+    host = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = lib.mdDoc ''
+        The host address that Audiobookshelf will listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8000;
+      description = lib.mdDoc ''
+        The port that Audiobookshelf will listen on.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "audiobookshelf";
+      description = lib.mdDoc ''
+        User account under which Audiobookshelf runs.
+      '';
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "audiobookshelf";
+      description = lib.mdDoc ''
+        Group under which Audiobookshelf runs.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to open the firewall for the port in {option}`services.audiobookshelf.port`.
+      '';
+    };
+
+    configDir = mkOption {
+      type = types.str;
+      default = "/etc/audiobookshelf";
+      description = lib.mdDoc ''
+        Configuration directory Audiobookshelf will use.
+      '';
+    };
+
+    metadataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/audiobookshelf";
```

Comment:
Similarly might be better to change this to `/var/lib/audiobookshelf/metadata`.

Reply:
Similarly, I agree that the metadata directory should be `/var/lib/audiobookshelf/metadata`, but see my other comment about the option of doing this implicitly through dataDir instead of explicitly.